app/PaymentDrivers/Stripe/BACS.php

```diff
@@ -0,0 +1,191 @@
+<?php
+
+/**
+ * Invoice Ninja (https://invoiceninja.com).
+ *
+ * @link https://github.com/invoiceninja/invoiceninja source repository
+ *
+ * @copyright Copyright (c) 2022. Invoice Ninja LLC (https://invoiceninja.com)
+ *
+ * @license https://www.elastic.co/licensing/elastic-license
+ */
+
+namespace App\PaymentDrivers\Stripe;
+
+use App\Exceptions\PaymentFailed;
+use App\Http\Requests\ClientPortal\Payments\PaymentResponseRequest;
+use App\Jobs\Util\SystemLogger;
+use App\Models\ClientGatewayToken;
+use App\Models\GatewayType;
+use App\Models\Payment;
+use App\Models\PaymentType;
+use App\Models\SystemLog;
+use App\PaymentDrivers\StripePaymentDriver;
+use App\PaymentDrivers\Stripe\Jobs\UpdateCustomer;
+use Stripe\Checkout\Session;
+use Stripe\PaymentIntent;
+use Stripe\PaymentMethod;
+use App\Utils\Number;
+
+class BACS
+{
+    public $stripe;
+
+    public function __construct(StripePaymentDriver $stripe)
+    {
+        $this->stripe = $stripe;
+    }
+
+    public function authorizeView(array $data)
+    {
+        $customer = $this->stripe->findOrCreateCustomer();
+        $data['session'] = Session::create([
+            'payment_method_types' => ['bacs_debit'],
+            'mode' => 'setup',
+            'customer' => $customer->id,
+            'success_url' => str_replace("%7B", "{", str_replace("%7D", "}", $this->buildAuthorizeUrl())),
+            'cancel_url' => route('client.payment_methods.index'),
+        ]);
+        return render('gateways.stripe.bacs.authorize', $data);
+    }
+    private function buildAuthorizeUrl(): string
+    {
+        return route('client.payment_methods.confirm', [
+            'method' => GatewayType::BACS,
+            'session_id' => "{CHECKOUT_SESSION_ID}",
+        ]);
+    }
+
+    public function authorizeResponse($request)
+    {
+        $this->stripe->init();
+        if ($request->session_id) {
+            $session = $this->stripe->stripe->checkout->sessions->retrieve($request->session_id, ['expand' => ['setup_intent']]);
```

Comment:
you may need to use the stripe 

Session::retrieve( method here, this current request is missing some important parameters

For example when retrieving a payment intent we pass a second array of options which include the idempotency_key and also the stripe connect key if one exists.

```php
PaymentIntent::retrieve($state['server_response']->id, array_merge($this->stripe->stripe_connect_auth, ['idempotency_key' => uniqid("st",true)]));
```

Reply:
I'm unsure about this one. According to the Stripe documentation, it's correct

```
// See your keys here: https://dashboard.stripe.com/apikeys
$stripe = new \Stripe\StripeClient('sk_test_Y17KokhC3SRYCQTLYiU5ZCD2');

$stripe->checkout->sessions->retrieve('{{SESSION_ID}}', ['expand' => ['setup_intent']]);
```